com.unity.render-pipelines.high-definition/Runtime/Water/ShaderPassWaterForward.hlsl

```diff
@@ -0,0 +1,210 @@
+#if defined(HAVE_VERTEX_MODIFICATION)
+    ApplyVertexModification(input, normalWS, positionRWS, _TimeParameters.xyz);
+#endif
+
+    // Output to the fragment
+    output.positionCS = TransformWorldToHClip(positionRWS);
+    output.positionRWS = positionRWS;
+    output.normalWS = normalWS;
+    output.texCoord0 = uv0;
+    output.texCoord1 = uv1;
+    return output;
+}
+
+struct PackedVaryingsToPS
+{
+    PackedVaryingsMeshToPS vmesh;
+
+    // SGVs must be packed after all non-SGVs have been packed.
+    // If there are several SGVs, they are packed in the order of HLSL declaration.
+
+    UNITY_VERTEX_OUTPUT_STEREO
+
+#if defined(PLATFORM_SUPPORTS_PRIMITIVE_ID_IN_PIXEL_SHADER) && SHADER_STAGE_FRAGMENT
+#if (defined(VARYINGS_NEED_PRIMITIVEID) || (SHADERPASS == SHADERPASS_FULL_SCREEN_DEBUG))
+    uint primitiveID : SV_PrimitiveID;
+#endif
+#endif
+
+#if defined(VARYINGS_NEED_CULLFACE) && SHADER_STAGE_FRAGMENT
+    FRONT_FACE_TYPE cullFace : FRONT_FACE_SEMANTIC;
+#endif
+};
+
+PackedVaryingsToPS PackVaryingsToPS(VaryingsToPS input)
+{
+    PackedVaryingsToPS output;
+    output.vmesh = PackVaryingsMeshToPS(input.vmesh);
+    UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(output);
+    return output;
+}
+
+FragInputs UnpackVaryingsToFragInputs(PackedVaryingsToPS packedInput)
+{
+    FragInputs input = UnpackVaryingsMeshToFragInputs(packedInput.vmesh);
+
+#if defined(PLATFORM_SUPPORTS_PRIMITIVE_ID_IN_PIXEL_SHADER) && SHADER_STAGE_FRAGMENT
+#if (defined(VARYINGS_NEED_PRIMITIVEID) || (SHADERPASS == SHADERPASS_FULL_SCREEN_DEBUG))
+    input.primitiveID = packedInput.primitiveID;
+#endif
+#endif
+
+#if defined(VARYINGS_NEED_CULLFACE) && SHADER_STAGE_FRAGMENT
+    input.isFrontFace = IS_FRONT_VFACE(packedInput.cullFace, true, false);
+#endif
+
+    return input;
+}
+
+PackedVaryingsToPS Vert(AttributesMesh inputMesh)
+{
+    VaryingsToPS varyingsType;
+    varyingsType.vmesh = VertMeshWater(inputMesh);
+    return PackVaryingsToPS(varyingsType);
+}
+
+void Frag(PackedVaryingsToPS packedInput, out float4 outColor : SV_Target0)
+{
+    UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(packedInput);
+    FragInputs input = UnpackVaryingsToFragInputs(packedInput);
+
+    // Compute the tile index
+    uint2 tileIndex = uint2(input.positionSS.xy) / GetTileSize();
+
+    // input.positionSS is SV_Position
+    PositionInputs posInput = GetPositionInput(input.positionSS.xy, _ScreenSize.zw, input.positionSS.z, input.positionSS.w, input.positionRWS.xyz, tileIndex);
+
+#ifdef VARYINGS_NEED_POSITION_WS
+    float3 V = GetWorldSpaceNormalizeViewDir(input.positionRWS);
+#else
+    // Unused
+    float3 V = float3(1.0, 1.0, 1.0); // Avoid the division by 0
+#endif
+
+    // Get the surface and built in data
+    SurfaceData surfaceData;
+    BuiltinData builtinData;
+    GetSurfaceAndBuiltinData(input, V, posInput, surfaceData, builtinData);
+
+    // Light layers need to be set manually here as there is no mesh renderer
+    builtinData.renderingLayers = DEFAULT_LIGHT_LAYERS;
```

Comment:
Ah good point, light layers will need to works with water, so we will need to add them on the water volume control or the equivalent mesh